src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -45,13 +45,56 @@ class ProductCard extends Component {
         return variantThumbnail || thumbnail;
     }
 
+    addOrConfigureProduct(variantIndex, linkTo) {
+        const { customFilters, product: { url_key, variants, type_id } } = this.props;
+
+        if (variants && type_id === 'configurable') {
+            const correctVariants = variants.reduce((correctVariants, { product }) => {
+                const isCorrectVariant = Object.keys(customFilters).every(filterKey => (
+                    customFilters[ filterKey ].find(value => +value === product[ filterKey ])
+                ));
+
+                if (isCorrectVariant) correctVariants.push(product);
+
+                return correctVariants;
+            }, []);
+
+            const hasNoMoreOptions = correctVariants.length === 1;
+
+            if (!hasNoMoreOptions) {
```

Comment:
Let's use `hasNoMoreOptions` right away, no need for const, as it is not used later on.